stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/MessageListFragment.kt

```diff
@@ -40,18 +40,39 @@ import io.getstream.chat.android.ui.message.list.viewmodel.factory.MessageListVi
 @Suppress("MemberVisibilityCanBePrivate")
 public open class MessageListFragment : Fragment() {
 
-    protected val cid: String by lazy {
+    private val lazyThreadMode: LazyThreadSafetyMode = LazyThreadSafetyMode.NONE
+
+    /** A specific channel cid to be connected with the Stream channel. */
+    protected val cid: String by lazy(lazyThreadMode) {
         requireNotNull(requireArguments().getString(ARG_CHANNEL_ID)) { "Channel cid must not be null" }
     }
-    protected val themeResId: Int by lazy { requireArguments().getInt(ARG_THEME_RES_ID) }
-    protected val messageId: String? by lazy { requireArguments().getString(ARG_MESSAGE_ID) }
-    protected val showHeader: Boolean by lazy { requireArguments().getBoolean(ARG_SHOW_HEADER, false) }
 
-    protected val factory: MessageListViewModelFactory by lazy { MessageListViewModelFactory(cid, messageId) }
-    protected val messageListViewModel: MessageListViewModel by viewModels { factory }
+    /** A custom theme resource for the screen. */
+    protected val themeResId: Int by lazy(lazyThreadMode) { requireArguments().getInt(ARG_THEME_RES_ID) }
```

Comment:
I'm curious, does removing synchronization increase performance here?

My personal taste is that it's not necessary to extract `LazyThreadSafetyMode.NONE` enum to a separate `val`. Maybe code would be more readable if enum was applied directly to lazy, like `lazy(LazyThreadSafetyMode.NONE)`?. 

Another option would be to create new function like `fun unsafeLazy(initializer) = lazy(LazyThreadSafetyMode.NONE, initializer)`?

Reply:
Exactly! This does removing synchronization works internally that increase performance when initializing them.
I agree the `LazyThreadSafetyMode` does not need to be extracted separately. 👍

Reply:
Applied in 7a48d9131